sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp

```diff
@@ -169,16 +169,21 @@ Azure::Response<KeyPropertiesSinglePage> KeyClient::GetPropertiesOfKeysSinglePag
       },
       request.Path,
       request.Query);
+  KeyPropertiesSinglePage responsePage = std::move(response.Value);
+  responsePage.RawResponse = std::move(response.RawResponse);
+  responsePage.m_keyClient = std::make_shared<KeyClient>(*this);
+  responsePage.m_type = KeyPropertiesSinglePage::KeyPropertiesType::Keys;
```

Comment:
This is a strange way to construct the type.  I would expect a constructor with validation of the fields.

Reply:
sure